src/main/java/com/yahoo/bullet/rest/pubsub/PubSubService.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright 2017, Yahoo Inc.
+ *  Licensed under the terms of the Apache License, Version 2.0.
+ *  See the LICENSE file associated with the project for terms.
+ */
+package com.yahoo.bullet.rest.pubsub;
+
+import com.yahoo.bullet.pubsub.PubSubMessage;
+import org.springframework.stereotype.Service;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+@Service
+public class PubSubService {
+    private ConcurrentLinkedQueue<PubSubMessage> queries = new ConcurrentLinkedQueue<>();
+    private ConcurrentLinkedQueue<PubSubMessage> results = new ConcurrentLinkedQueue<>();
+
+    /**
+     * Get the next query from the query queue. Queries are removed after being read once.
+     *
+     * @return The next query.
+     */
+    public String getQuery() {
+        return queries.isEmpty() ? null : queries.poll().asJSON();
```

Comment:
Can we have a private method getFromQueue(ConcurrentLInkedQueue) that does this and use it for both methods?